File: app/src/main/java/com/meetferrytan/mvpdaggerstore/util/interceptors/MyRequestInterceptor.java
package com.meetferrytan.mvpdaggerstore.util.interceptors;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * Created by ferrytan on 12/21/16.
 */

// TODO implement your own request interceptor
public class MyRequestInterceptor implements Interceptor {
    private static final String HEADER_AUTHORIZATION = "Authorization";
    private static final String AUTH_PREFIX = "Bearer ";

    public MyRequestInterceptor() {
    }

    @Override
    public Response intercept(Chain chain) throws IOException {
        Request request = chain.request();
        Request newRequest;
/*
        String token = authToken;
        if(!"".equals(token)) {
            newRequest = request.newBuilder()
                    .addHeader(HEADER_AUTHORIZATION, AUTH_PREFIX + token)
                    .build();
            return chain.proceed(newRequest);
        }*/
        return chain.proceed(request);
    }
}

File: app/src/main/java/com/meetferrytan/mvpdaggerstore/util/interceptors/MyRequestInterceptor.kt
package com.meetferrytan.mvpdaggerstore.util.interceptors

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

/**
 * Created by ferrytan on 12/21/16.
 */

// TODO implement your own request interceptor
class MyRequestInterceptor : Interceptor {

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val newRequest: Request
        /*
        String token = authToken;
        if(!"".equals(token)) {
            newRequest = request.newBuilder()
                    .addHeader(HEADER_AUTHORIZATION, AUTH_PREFIX + token)
                    .build();
            return chain.proceed(newRequest);
        }*/
        return chain.proceed(request)
    }

    companion object {
        private val HEADER_AUTHORIZATION = "Authorization"
        private val AUTH_PREFIX = "Bearer "
    }
}